src/client/scripts/structures/user.js

```diff
@@ -8,15 +8,239 @@ export class User {
+     * @param {Number[]} following - stores those being followed by the User
+     * @param {Object} metadata - Object mapping Deck IDs to study information
+     * Study information is an object including three keys
+     * timeLastStudied - The time in milliseconds since the Epoch at the time the deck was last studied
+     * timesStudied - The number of times the deck has been studied, to be used for spatial repetition
+     * beingStudied - A boolean indicating whether the deck is being actively studied
      */
-   constructor(id = null, username = null, metadata = null, followers = null, following = null) {
+   constructor(id = null, username = null, followers = null, following = null, metadata = {}) {
       this.id = id;
       this.username = username;
-      this.metadata = metadata;
       this.followers = followers;
       this.following = following;
+      this.metadata = metadata;
+   }
+
+   /**
+    * An abstraction to get the active user from local storage. Unnecessary, but there if you want it.
+    * @returns {User} - The active (logged in) user object
+    */
+   static getActiveUser() {
+      return localStorage.getItem("active-user");
+   }
+
+   /**
+    * @todo - Update database after registry
+    * Registers a deck's id in the metadata field along with necessary information
+    * @param {Deck} deck - Deck object as defined in /structures/deck.js
+    */
+   registerDeck(deck) {
+      this.metadata[deck.id] = {};
+      this.metadata[deck.id].timeLastStudied = Date.now();
+      this.metadata[deck.id].timesStudied = 0;
+      this.metadata[deck.id].beingStudied = true;
+   }
+
+   /**
+    * Checks whether the user needs to study a given deck based on spatial repetition
+    * @param {Deck} deck - Deck object as defined in /structures/deck.js
+    * @returns {boolean} - true if the user needs to study this deck, false otherwise
+    */
+   checkDeck(deck) {
+      let time = Date.now();
+      let timeInDay = 86400000; // Number of milliseconds in a day
+      return ((time - this.metadata[deck.id].timeLastStudied) >= (timeInDay * this.metadata[deck.id].timesStudied));
+   }
+
+   /**
+    * @todo - Modify to update database
+    * Updates the given deck in metadata, to be used after the user studies it
+    * @param {Deck} deck - Deck object as defined in /structures/deck.js
+    */
+   updateDeck(deck) {
+      this.metadata[deck.id].timeLastStudied = Date.now();
+      this.metadata[deck.id].timesStudied += 1;
+   }
+
+   /**
+    * Utility method for comparing the current state of a user's deck list to local storage.
+    * Not needed elsewhere
+    * @param {Number[]} d1 - Array of Deck IDs
+    * @param {Number[]} d2 - Array of Deck IDs
+    * @returns 
+    */
+   _compareUserDecks(d1, d2) {
+      if (d1.length !== d2.length) {
+         return false;
+      }
+      for (let i = 0; i < d1.length; i++) {
+         if (d1[i] !== d2[i]) {
+            return false;
+         }
+      }
+      return true
+   }
```

Comment:
This will work for the most part, but not in a scenario where a deck's properties are modified.
- User decks get stored into localStorage
- User modifies one of their decks
- _compareUserDecks(d1,d2) returns true, but the state has actually changed

That being said, persisting the state of the current decks may not be an issue (this is more of a milestone03 thing imo). We could keep this in for now and maybe change it later.